si2.bll/Dtos/Requests/ProgramLevel/UpdateProgramLevelDto.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Text;
+
+namespace si2.bll.Dtos.Requests.ProgramLevel
+{
+    public class UpdateProgramLevelDto
+    {
+        [Required]
+        [RegularExpression(@"^\d+\.\d{0,2}$", ErrorMessage = "Please enter valid float Number")]
+        [Range(0, 999.99)]
+        public float Credits { get; set; }
+
+        [Required(ErrorMessage = "NameFr field is mandatory")]
+        [StringLength(30, ErrorMessage = "NameFr field must be equal or below 30 characters")]
+        public string NameFr { get; set; }
+
+        [Required(ErrorMessage = "NameAr field is mandatory")]
+        [StringLength(30, ErrorMessage = "NameAr field must be equal or below 30 characters")]
+        public string NameAr { get; set; }
+
+        [Required(ErrorMessage = "NameEn field is mandatory")]
+        [StringLength(30, ErrorMessage = "NameEn field must be equal or below 30 characters")]
+        public string NameEn { get; set; }
```

Comment:
I suggest to remove the error messages let the system generate them
Because :NameEn is not good it should be the ENglish Name of ...